pkg/autoscaler/scaling/autoscaler.go

```diff
@@ -245,8 +244,49 @@ func (a *Autoscaler) Scale(ctx context.Context, now time.Time) (desiredPodCount
 	return desiredPodCount, int32(excessBCF), true
 }
 
+func clampInclusiveRange(valueToClamp float64, lowerBound float64, upperBound float64) float64 {
+	return math.Min(math.Max(valueToClamp, lowerBound), upperBound)
+}
+
+// Query functions
+func (a *Autoscaler) currentlyStable() bool {
+	return a.panicTime.IsZero()
+}
+
+func (a *Autoscaler) currentlyPanicking() bool {
+	return !a.currentlyStable()
+}
+
+func (a *Autoscaler) stableWindowHasElapsed(spec *DeciderSpec, now time.Time) bool {
```

Comment:
probably more useful would be `hasElapsed(time, duration)` and pass `spec.StableWindow` in the call site.

Reply:
I'm inclined to keep this as-is, there's only one callsite and I wanted to emphasise the full predicate.

Reply:
but passing spec.StableWindow  to has elapsed is actually is more expressive (or not less)

Reply:
Ah, I get you now. I was thinking of the name of the method rather than the `spec` parameter type being silly for a single use.

Reply:
Updated in 97e1298.